app/services/NotProcessedFilesService.scala

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package services
+
+import config.AppConfig
+import javax.inject.Inject
+import models.FailedJobResponses.FailedToProcessNotifications
+import models.MongoLockResponses
+import models.notification.RecordStatusEnum
+import play.api.Configuration
+import repositories.FileNotificationRepository
+import scheduler.ScheduleStatus.JobFailed
+import scheduler.{ScheduleStatus, ScheduledService}
+import uk.gov.hmrc.mongo.lock.{LockRepository, LockService, MongoLockRepository}
+import utils.Logger.logger
+import utils.PagerDutyHelper.PagerDutyKeys.{FAILED_TO_PROCESS_FILE_NOTIFICATION, MONGO_LOCK_UNKNOWN_EXCEPTION, NOTIFICATION_SET_TO_NOT_PROCESSED_PENDING_RETRY, UNKNOWN_EXCEPTION_FROM_SDES}
+import utils.{PagerDutyHelper, TimeMachine}
+
+import scala.concurrent.duration.{Duration, DurationInt}
+import scala.concurrent.{ExecutionContext, Future}
+
+class NotProcessedFilesService @Inject()(lockRepositoryProvider: MongoLockRepository,
+                                         fileNotificationRepository: FileNotificationRepository,
+                                         timeMachine: TimeMachine,
+                                         config: Configuration,
+                                         appConfig: AppConfig
+                                        )(implicit ec: ExecutionContext) extends ScheduledService[Either[ScheduleStatus.JobFailed, String]] {
+
+  val jobName = "NotProcessedFilesService"
+  lazy val mongoLockTimeoutSeconds: Int = config.get[Int](s"schedules.$jobName.mongoLockTimeout")
+
+  lazy val lockKeeper: LockService = new LockService {
+    override val lockId: String = s"schedules.$jobName"
+    override val ttl: Duration = mongoLockTimeoutSeconds.seconds
+    override val lockRepository: LockRepository = lockRepositoryProvider
+  }
+
+  //scalastyle:off
+  override def invoke: Future[Either[JobFailed, String]] = {
+    tryLock {
+      logger.info(s"[$jobName][invoke] - Job started")
+      for {
+        filesReceived <- fileNotificationRepository.getFilesReceivedBySDES()
+        filteredFiles = {
+          logger.info(s"[NotProcessedFilesService][invoke] - Number of files received by SDES: ${filesReceived.size}")
```

Comment:
This seems a bit misleading, could we change it to be something like:
```suggestion
          logger.info(s"[NotProcessedFilesService][invoke] - Number of records in $FileReceivedInSDES state: ${filesReceived.size}")
```